src/quick-lint-js/diag/diagnostic-types-2.h

```diff
@@ -3538,6 +3538,13 @@ struct Diag_Xor_Used_As_Exponentiation {
   Source_Code_Span xor_operator;
 };
 
+struct Diag_Typeof_Variable_Equals_Undefined {
+  [[qljs::diag("E0458", Diagnostic_Severity::warning)]] //
+  [[qljs::message("typeof result is of type string and so will never equal "
+                  "undefined; use 'undefined' instead", ARG(typeof_operator))]] //
```

Comment:
Instead of highlighting the `typeof` operator, I think we should highlight the `undefined` variable. We try to highlight the thing that the user needs to change. (You can keep the `typeof_operator` variable, but leave it unused.)
```suggestion
  [[qljs::message("typeof result is of type string and so will never equal "
                  "undefined; use 'undefined' instead", ARG(undefined))]] //
```